source/agora/common/Set.d

```diff
@@ -51,11 +56,33 @@ public struct Set (T)
     /// https://github.com/dlang/druntime/pull/3412
     public string toString () const
     {
-        import std.format;
+        string ret;
+        scope SinkT dg = (scope v) { ret ~= v; };
+        this.toString(dg);
+        return ret;
+    }
+
+    /// Ditto
+    public void toString (scope void delegate(const(char)[]) @safe sink) const
+    {
+        import std.format : formattedWrite;
 
-        return format("%(%s, %)",
-                      this._set.byKeyValue.filter!(tup => tup.value)
-                      .map!(tup => tup.key));
+        formattedWrite(sink, "[%(%s, %)]", this._set.byKeyValue().filter!(
+            kv => kv.value).map!(kv => kv.key));
```

Comment:
Why is one side using a policy and the other side is not ?

Reply:
The `deserializeWithPolicy` method requires a policy as a parameter, but the contrary `serializeWithPolicy`  method was found to be not necessary for the `toString`.

Should it be used here?